app/src/main/java/com/example/makeitso/screens/login/LoginViewModel.kt

```diff
@@ -62,33 +60,8 @@ class LoginViewModel @Inject constructor(
             return
         }
 
-        viewModelScope.launch(showErrorExceptionHandler) {
-            val oldUserId = accountService.getUserId()
-            accountService.authenticate(email, password) { error ->
-                if (error == null) {
-                    linkWithEmail()
-                    updateUserId(oldUserId, openAndPopUp)
-                } else onError(error)
-            }
-        }
-    }
-
-    private fun linkWithEmail() {
-        viewModelScope.launch(showErrorExceptionHandler) {
-            accountService.linkAccount(email, password) { error ->
-                if (error != null) logService.logNonFatalCrash(error)
-            }
-        }
-    }
-
-    private fun updateUserId(oldUserId: String, openAndPopUp: (String, String) -> Unit) {
-        viewModelScope.launch(showErrorExceptionHandler) {
-            val newUserId = accountService.getUserId()
-
-            storageService.updateUserId(oldUserId, newUserId) { error ->
-                if (error != null) logService.logNonFatalCrash(error)
-                else openAndPopUp(SETTINGS_SCREEN, LOGIN_SCREEN)
-            }
+        launchCatching {
+            accountService.authenticate(email, password)
```

Comment:
We still need to call `openAndPopUp(SETTINGS_SCREEN, LOGIN_SCREEN)` after authenticating, can you add this?

Reply:
good catch, added